lib/libutil/getlocalbase.c

```diff
@@ -38,8 +38,12 @@ __FBSDID("$FreeBSD$");
 #include <unistd.h>
 
 #ifndef _PATH_LOCALBASE
+#ifdef COMPAT_64BIT
+#define	_PATH_LOCALBASE	"/usr/local64"
+#else
 #define _PATH_LOCALBASE "/usr/local"
```

Comment:
@bsdjhb had an interesting point on how we should define the local base. He suggested to always use `/usr/local64` for freebsd64 binaries and always use `/usr/local64c` for freebsd64c binaries. Otherwise we would have several cases to handle, in a pure-capability world we would have to place freebsd64c binaries in `/usr/local`, freebsd64 binaries in `/usr/local64`, and in a hybrid world we would have to place freebsd64c binaries in `/usr/local64c`, freebsd64 binaries in `/usr/local`, but most importantly we would have to build a single program (from the base system or ports) available for freebsd64 and freebsdc four times: {freebsd64, freebsd64c}x{pure-capability world, hybrid world}.

I agree with John's point and hence we should define `_PATH_LOCALBASE` as `/usr/local64c` for pure-capability binaries and as `/usr/local64` when native binaries are freebsd64, at least for now.

Reply:
If you agree with this, then we should also update /etc/mtree to reflect this change.

Reply:
I disagree, this is a waste of time that deviates more from upstream FreeBSD behaviour, as well as how we structure (/usr)/lib. Hybrid worlds exist solely for testing freebsd64, they are of no interest outside of that and will never be something we ship to users. You can't even build one with cheribuild by default any more unless you pass a flag to enable it.

Reply:
@jrtc27 Does it mean you don't want to build packages for the hybrid world, freebsd64 nor freebsd64c?

Reply:
OK. Having this assumption, should we always use `/usr/local` and `/usr/local64`, including the hybrid world?

I haven't tested that but, if I understand the code correctly, `LIB64` is marked as broken on non-pure-capability targets, in which case `COMPAT_64BIT` is not defined as well. If we agree with the above, we must define `_PATH_LOCALBASE` as `"/usr/local64"` also when `COMPAT_64BIT` is not defined and the world is hybrid.

Reply:
/usr/local64 won't exist in a hybrid world, and shouldn't, because that's the native userspace ABI. As I said, we really do not care about hybrid worlds, let alone providing packages for them. For those who go ahead and jump through all the hoops to build such a thing manually themselves, if they really want packages, they can just grab normal FreeBSD AArch64 ones.

Reply:
OK.